packages/react-native-reanimated/src/layoutReanimation/web/componentUtils.ts

```diff
@@ -228,12 +235,108 @@ export function handleLayoutTransition(
       break;
   }
 
-  animationConfig.animationName = TransitionGenerator(
-    animationType,
-    transitionData
+  const { transitionKeyframeName, dummyTransitionKeyframeName } =
+    TransitionGenerator(animationType, transitionData);
+
+  if (animationType !== TransitionType.CURVED) {
+    animationConfig.animationName = transitionKeyframeName;
+    setElementAnimation(element, animationConfig);
+  } else {
+    handleCurvedTransition(
+      element,
+      animationConfig,
+      transitionData,
+      transitionKeyframeName,
+      dummyTransitionKeyframeName! // In `CurvedTransition` it cannot be undefined
+    );
+  }
+}
+
+function handleCurvedTransition(
```

Comment:
This function may be moved into `Curved.web.ts`, let me know what do you think @piaskowyk 😅